### PR TITLE
[Dataset] PPIDataset

### PR DESCRIPTION
**Pull request by HuXiangkun**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: examples/pytorch/cluster_gcn/cluster_gcn.py [hunk 1]
```diff
@@ -138,8 +136,8 @@ def main(args):
     # set train_nids to cuda tensor
     if cuda:
         train_nid = torch.from_numpy(train_nid).cuda()
-    print("current memory after model before training",
-          torch.cuda.memory_allocated(device=train_nid.device) / 1024 / 1024)
+        print("current memory after model before training",
```
Comment (hetong007) [inline examples/pytorch/cluster_gcn/cluster_gcn.py:139]: control `print` with `self.verbose`

File: python/dgl/data/ppi.py [hunk 2]
```diff
@@ -60,18 +66,20 @@ def __init__(self, mode='train', raw_dir=None, force_reload=False, verbose=False
...
+        graph_file = os.path.join(self.save_path, '{}_graph.json'.format(self.mode))
+        label_file = os.path.join(self.save_path, '{}_labels.npy'.format(self.mode))
+        feat_file = os.path.join(self.save_path, '{}_feats.npy'.format(self.mode))
+        graph_id_file = os.path.join(self.save_path, '{}_graph_id.npy'.format(self.mode))
 
         g_data = json.load(open(graph_file))
         self._labels = np.load(label_file)
         self._feats = np.load(feat_file)
-        self.graph = DGLGraph(nx.DiGraph(json_graph.node_link_graph(g_data)))
+        self._graph = DGLGraph(nx.DiGraph(json_graph.node_link_graph(g_data)))
```
Comment (hetong007) [inline python/dgl/data/ppi.py:78]: if `graph` is a property, why not keep using it?

File: examples/pytorch/cluster_gcn/cluster_gcn.py (hunk 1 above)
Comment (HuXiangkun) [reply examples/pytorch/cluster_gcn/cluster_gcn.py:139]: This code is in example, and it's a bug fix. When running with CPU, this code failed.

File: python/dgl/data/ppi.py (hunk 2 above)
Comment (HuXiangkun) [reply python/dgl/data/ppi.py:78]: Moved it to @property

Comment (hetong007) [reply python/dgl/data/ppi.py:78]: I mean can we keep using `self.graph` instead of `self._graph`

File: examples/pytorch/cluster_gcn/README.md [hunk 3]
```diff
@@ -24,7 +25,8 @@ Dependencies
 2) cd metis-5.1.0
 3) make config shared=1 prefix=~/.local/
 4) make install
-5) export METIS_DLL=~/.local/lib/libmetis.so
+5) export METIS_DLL=~/.local/lib/libmetis.so
```
Comment (classicsong) [inline examples/pytorch/cluster_gcn/README.md:28]: metis is builtin right now.

File: python/dgl/data/ppi.py [hunk 4]
```diff
@@ -83,34 +91,42 @@ def process(self, root_path):
         for g_id in range(lo, hi):
             g_mask = np.where(graph_id == g_id)[0]
             graph_masks.append(g_mask)
-            g = self.graph.subgraph(g_mask)
+            g = self._graph.subgraph(g_mask)
```
Comment (classicsong) [inline python/dgl/data/ppi.py:94]: Use self.graph.subgraph(g_mask)

File: python/dgl/data/ppi.py [hunk 5]
```diff
@@ -91,7 +91,7 @@ def process(self):
         for g_id in range(lo, hi):
             g_mask = np.where(graph_id == g_id)[0]
             graph_masks.append(g_mask)
-            g = self._graph.subgraph(g_mask)
+            g = self.graph.subgraph(g_mask)
             g.ndata['feat'] = F.tensor(self._feats[g_mask], dtype=F.data_type_dict['float32'])
             g.ndata['label'] = F.tensor(self._labels[g_mask], dtype=F.data_type_dict['int64'])
             self._graphs.append(g)
```
Comment (hetong007) [inline python/dgl/data/ppi.py:97]: `self._graphs` and `self.graphs` are still mismatching (here and there).

File: examples/pytorch/cluster_gcn/README.md (hunk 3 above)
Comment (classicsong) [reply examples/pytorch/cluster_gcn/README.md:28]: I think we donot need this in the README.md

**Pull request merged**